src/josepy/json_util.py

```diff
@@ -233,7 +237,17 @@ def __new__(
                 if isinstance(value, _TypedField):
                     # Ensure the type annotation has been set for the field.
                     # Error out if it is not the case.
-                    if key not in namespace.get("__annotations__", {}):
+                    if "__annotations__" in namespace:
+                        annotations = namespace["__annotations__"]
+                    elif sys.version_info >= (3, 14) and (
+                        annotate := annotationlib.get_annotate_from_class_namespace(namespace)
+                    ):
+                        annotations = annotationlib.call_annotate_function(
+                            annotate, format=annotationlib.Format.FORWARDREF
+                        )
```

Comment:
I personally think use of walrus operators should be limited, and certainly if statements should not be complex enough to require multiple lines. What about the following:

```suggestion
                    elif sys.version_info >= (3, 14):
                        annotate = annotationlib.get_annotate_from_class_namespace(namespace)
                        if annotate: 
                            annotations = annotationlib.call_annotate_function(
                                annotate, format=annotationlib.Format.FORWARDREF
                            )
                        else:
                            annotations = {}
```

This seems more readable to me, but perhaps I'm being curmudgeonly about new language features here.

Reply:
Not sure that's an improvement. To fully capture all options, you'd need to `else` clauses. So overall the `:=` use makes it more readable IMO. `:=` is also used in the example from the Python docs I've adapted this section from.
https://docs.python.org/3.14/library/annotationlib.html#using-annotations-in-a-metaclass

```py
if "__annotations__" in namespace:
    annotations = namespace["__annotations__"]
elif sys.version_info >= (3, 14):
    annotate = annotationlib.get_annotate_from_class_namespace(namespace)
    if annotate: 
        annotations = annotationlib.call_annotate_function(
            annotate, format=annotationlib.Format.FORWARDREF
        )
    else:
        annotations = {}
else:
    annotations = {}
```

Reply:
Like list comprehensions, walrus operators should only be used in simple, readable cases. This case differs from the python example because the original `elif` is checking multiple things, creating an overly long expression to check. I think that code block directly above looks great, as it's what I suggested above. It makes it very clear what happens in the various cases, and in the future when only 3.14+ is available, it will be very neat and clear to update.

If you're really excited about walruses and don't like else statements, the following is also tolerable:

```py
annotations = {}
if "__annotations__" in namespace:
    annotations = namespace["__annotations__"]
elif sys.version_info >= (3, 14):
    if annotate := annotationlib.get_annotate_from_class_namespace(namespace):
        annotations = annotationlib.call_annotate_function(
            annotate, format=annotationlib.Format.FORWARDREF
        )
```

or, without the walrus:

```py
annotations = {}
if "__annotations__" in namespace:
    annotations = namespace["__annotations__"]
elif sys.version_info >= (3, 14):
    annotate = annotationlib.get_annotate_from_class_namespace(namespace)
    if annotate: 
        annotations = annotationlib.call_annotate_function(
            annotate, format=annotationlib.Format.FORWARDREF
        )
```

My main issue here is the length and complexity of the expression to be evaluated in the `elif`. Solutions that avoid that problem work for me.